image-hd.c

```diff
@@ -671,9 +698,13 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 			return -EINVAL;
 		}
 		if (part->offset && part->in_partition_table) {
-			if (now > part->offset) {
+			if (!part->extended) {
+				int ret = check_overlap(image, part);
+				if (ret)
+					return ret;
+			} else if (now > part->offset) {
 				image_error(image, "part %s overlaps with previous partition\n",
-						part->name);
+					    part->name);
```

Comment:
Unrelated whitespace change. The whole file is indented inconsistently. Sometimes 2 tabs, sometimes aligned with the `(`. We should clean that up, but not mixed into other commits.

Reply:
fixed (though the code gets moved and outdented in a later commit anyway).